plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,834 @@
+                match.details["cdp"]["adminState"] = cdp_admin_state
+
+            if pfc_admin_state and match.details.get("pfc", {}).get("adminState") != pfc_admin_state:
+                ops.append(dict(op="replace", path="{0}/{1}/pfc/adminState".format(path, match.index), value=pfc_admin_state))
+                match.details["pfc"]["adminState"] = pfc_admin_state
+
+            if llfc_transmit_state and match.details.get("llfc", {}).get("transmitState") != llfc_transmit_state:
+                ops.append(dict(op="replace", path="{0}/{1}/llfc/transmitState".format(path, match.index), value=llfc_transmit_state))
+                match.details["llfc"]["transmitState"] = llfc_transmit_state
+
+            if llfc_receive_state and match.details.get("llfc", {}).get("receiveState") != llfc_receive_state:
+                ops.append(dict(op="replace", path="{0}/{1}/llfc/receiveState".format(path, match.index), value=llfc_receive_state))
+                match.details["llfc"]["receiveState"] = llfc_receive_state
+
+            if stp_bpdu_filter and match.details.get("stp", {}).get("bpduFilterEnabled") != stp_bpdu_filter:
+                ops.append(dict(op="replace", path="{0}/{1}/stp/bpduFilterEnabled".format(path, match.index), value=stp_bpdu_filter))
+                match.details["stp"]["bpduFilterEnabled"] = stp_bpdu_filter
+
+            if stp_bpdu_guard and match.details.get("stp", {}).get("bpduGuardEnabled") != stp_bpdu_guard:
+                ops.append(dict(op="replace", path="{0}/{1}/stp/bpduGuardEnabled".format(path, match.index), value=stp_bpdu_guard))
+                match.details["stp"]["bpduGuardEnabled"] = stp_bpdu_guard
+
+            if l2_interface_qinq and match.details.get("l2Interface", {}).get("qinq") != l2_interface_qinq:
+                ops.append(dict(op="replace", path="{0}/{1}/l2Interface/qinq".format(path, match.index), value=l2_interface_qinq))
+                match.details["l2Interface"]["qinq"] = l2_interface_qinq
+
+            if l2_interface_reflective_relay and match.details.get("l2Interface", {}).get("reflectiveRelay") != l2_interface_reflective_relay:
+                ops.append(dict(op="replace", path="{0}/{1}/l2Interface/reflectiveRelay".format(path, match.index), value=l2_interface_reflective_relay))
+                match.details["l2Interface"]["reflectiveRelay"] = l2_interface_reflective_relay
+
+            if vlan_scope and match.details.get("l2Interface", {}).get("vlanScope") != vlan_scope:
+                ops.append(dict(op="replace", path="{0}/{1}/l2Interface/vlanScope".format(path, match.index), value=vlan_scope))
+                match.details["l2Interface"]["vlanScope"] = vlan_scope
+
+            if lldp:
+                # if LLDP status is disabled, then receive_state and transmit_state must be disabled
+                if lldp["status"] == "disabled" and not (lldp["receive_state"] == "disabled" and lldp["transmit_state"] == "disabled"):
```

Comment:
```suggestion
                if lldp.get("status")== "disabled" and not (lldp.get("receive_state") == "disabled" and lldp.get("transmit_state") == "disabled"):
```
this `get()` method must also be applied for the rest of existence conditions for `lldp` and `mcp` for update and creation checks

Reply:
As discussed before, since it is not affected by the current behavior of ansible, this change will not be made.